### PR TITLE
Adding elastichosts Dallas and Miami providers

### PR DESCRIPTION
**Pull request by stuarthendren**
These are just based on the existing ones, and may need some tweaking, say to the templates.

### CODE REVIEW

File: providers/elastichosts-dal-a/pom.xml [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  <description>ElasticHosts implementation targeted to Dallas</description>
+  <packaging>bundle</packaging>
+
+  <properties>
+    <test.elastichosts-ams-e.endpoint>https://api-dal-a.elastichosts.com</test.elastichosts-ams-e.endpoint>
+    <test.elastichosts-ams-e.api-version>2.0</test.elastichosts-ams-e.api-version>
+    <test.elastichosts-ams-e.build-version />
+    <test.elastichosts-ams-e.identity>FIXME_IDENTITY</test.elastichosts-ams-e.identity>
+    <test.elastichosts-ams-e.credential>FIXME_CREDENTIAL</test.elastichosts-ams-e.credential>
+    <test.elastichosts-ams-e.template />
```
Comment (nacx) [inline providers/elastichosts-dal-a/pom.xml]: Change the test properties to match the provider id

File: providers/elastichosts-dal-a/src/main/java/org/jclouds/elastichosts/ElasticHostsDallasMetadata.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import java.util.Properties;
+
+import org.jclouds.elasticstack.ElasticStackApiMetadata;
+import org.jclouds.providers.ProviderMetadata;
+import org.jclouds.providers.internal.BaseProviderMetadata;
+
+import com.google.auto.service.AutoService;
+
+/**
+ * Implementation of {@link ProviderMetadata} for ElasticHosts Amsterdam.
```
Comment (nacx) [inline providers/elastichosts-dal-a/src/main/java/org/jclouds/elastichosts/ElasticHostsDallasMetadata.java]: Fix javadoc

File: providers/elastichosts-mmi-a/pom.xml [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  <description>ElasticHosts implementation targeted to Miami</description>
+  <packaging>bundle</packaging>
+
+  <properties>
+    <test.elastichosts-ams-e.endpoint>https://api-mmi-a.elastichosts.com</test.elastichosts-ams-e.endpoint>
+    <test.elastichosts-ams-e.api-version>2.0</test.elastichosts-ams-e.api-version>
+    <test.elastichosts-ams-e.build-version />
+    <test.elastichosts-ams-e.identity>FIXME_IDENTITY</test.elastichosts-ams-e.identity>
+    <test.elastichosts-ams-e.credential>FIXME_CREDENTIAL</test.elastichosts-ams-e.credential>
+    <test.elastichosts-ams-e.template />
```
Comment (nacx) [inline providers/elastichosts-mmi-a/pom.xml]: Same here. Fix test property names.

**Comment by nacx** [comment]
The build succeeded; the failure was after the build when Jenkins was collecting data.
Thanks @stuarthendren! Are you able to run the live tests against the new regions?

**Comment by stuarthendren** [comment]
I have ran the live tests and I get some test failures. 

```
Results :

Failed tests:
  ElasticHostsMiamiApiLiveTest>ElasticStackApiLiveTest.testGetDrive:140 » HttpResponse
  ElasticHostsMiamiApiLiveTest>ElasticStackApiLiveTest.testListDrives:114 » HttpResponse
  ElasticHostsMiamiApiLiveTest>ElasticStackApiLiveTest.testListStandardDrives:120 » HttpResponse
  ElasticHostsMiamiTemplateBuilderLiveTest.testDefaultTemplateBuilder:43 expected [true] but found [false]
  ElasticHostsMiamiComputeServiceLiveTest>BaseComputeServiceLiveTest.testCreateAndRunAService:708->BaseComputeServiceLiveTest.createAndRunAServiceInGroup:746->BaseComputeServiceLiveTest .exec:757->BaseComputeServiceLiveTest.exec:761 » IllegalArgument
  ElasticHostsMiamiApiLiveTest>ElasticStackApiLiveTest.testCreateAndStartServer:199->ElasticStackApiLiveTest.prepareDrive:360 » HttpResponse
  ElasticHostsMiamiComputeServiceLiveTest>BaseComputeServiceLiveTest.testAScriptExecutionAfterBootWithBasicTemplate:262->BaseComputeServiceLiveTest.weCanCancelTasks:294 » IllegalArgument
  ElasticHostsMiamiComputeServiceLiveTest>BaseComputeServiceLiveTest.testCreateTwoNodesWithRunScript:349->BaseComputeServiceLiveTest.checkNodes:510->BaseComputeServiceLiveTest.sshPing:880->BaseComputeServiceLiveTest.doCheckJavaIsInstalledViaSsh:893 » IllegalArgument

Tests run: 50, Failures: 8, Errors: 0, Skipped: 15
```

However, I also get failures when running against a current elastichosts zone, eg lon-b. So I may be configuring the tests incorrectly. I am simply supping the `test.elastichosts-mmi-a.identity` and `test.elastichosts-mmi-a.credential` variables, and just using the defaults for the others. If I am doing it wrong if I could get a valid set of settings (minus identity and credential) for lon-b I could try again on the new providers.

**Comment by nacx** [comment]
>However, I also get failures when running against a current elastichosts zone, eg lon-b. So I may be configuring the tests incorrectly

Out the tests are out of sync with new images or newer versions of the api. Could you share the test log with the stacktraces in a gist or pastie? Unfortunately I don't have an account I can use to test, but I'll try to have a look at them.

**Comment by stuarthendren** [comment]
One failure was the template check, which I have fixed and simplified the template.

I have also noticed that some fail due to lack of funds in my account, as I was using the trial and their must be restrictions on the size of machine you can create. I may get some funds at some point but this means I can't currently run the full set of tests. 

However, three tests fail that request Drive info, ElasticHostsMiamiApiLiveTest .testGetDrive, testListDrives and testListStandardDrives. They all fail to parse the response. See https://gist.github.com/stuarthendren/d8f633f0db2a0045f17b for stacktrace.

**Comment by nacx** [comment]
>I have also noticed that some fail due to lack of funds in my account, as I was using the trial and their must be restrictions on the size of machine you can create.

Same happens to me. Regarding the drives failure, can you also share the [jclouds.wire](http://jclouds.apache.org/reference/logging/) logs? This way we'll see the requests/responses being sent and we'll be able to see what happens. Also, could you create a JIRA issue to track this failure?

Thanks for your help @stuarthendren!

**Comment by stuarthendren** [comment]
Updated Gist with wire log for testCreateDrive and testGetDrive
https://gist.github.com/stuarthendren/d8f633f0db2a0045f17b

**Comment by stuarthendren** [comment]
Created Issue [JCLOUDS-914](https://issues.apache.org/jira/browse/JCLOUDS-914)

**Comment by nacx** [comment]
Pushed to [master](http://git-wip-us.apache.org/repos/asf/jclouds/commit/c3f17751) and [1.9.x](http://git-wip-us.apache.org/repos/asf/jclouds/commit/783a9aa5). Thanks @stuarthendren and apologies for the delay!

**Pull request closed without merging**